test_nested_parallel_limit fails on 1-core-VM

While working on reproducible builds for openSUSE, I noticed
that 2 tests would fail on 1-core-VMs (using python-joblib 0.12.1)

```
_______________________ test_nested_parallel_limit[loky] _______________________

>       assert backend_types_and_levels == expected_types_and_levels
E       AssertionError: assert [('LokyBacken...lBackend', 2)] == [('LokyBackend...lBackend', 3)]

____________________ test_nested_parallel_limit[threading] _____________________        
backend = 'threading'

    @with_multiprocessing
    @parametrize('backend', ['loky', 'threading'])
    def test_nested_parallel_limit(backend):
        with parallel_backend(backend, n_jobs=2):
            backend_types_and_levels = _recursive_backend_info()

        top_level_backend_type = backend.title() + 'Backend'
        expected_types_and_levels = [
            (top_level_backend_type, 0),
            ('ThreadingBackend', 1),
            ('SequentialBackend', 2),
            ('SequentialBackend', 3)
        ]
>       assert backend_types_and_levels == expected_types_and_levels
E       AssertionError: assert [('ThreadingB...lBackend', 2)] == [('ThreadingBa...lBackend', 3)]
```

--- a/joblib/_parallel_backends.py
+++ b/joblib/_parallel_backends.py
@@ -130,7 +130,6 @@ class ParallelBackendBase(with_metaclass(ABCMeta)):
         else:
             return ThreadingBackend(nesting_level=nesting_level)
 
-
     @contextlib.contextmanager
     def retrieval_context(self):
         """Context manager to manage an execution context.
@@ -348,7 +347,8 @@ class ThreadingBackend(PoolManagerMixin, ParallelBackendBase):
         n_jobs = self.effective_n_jobs(n_jobs)
         if n_jobs == 1:
             # Avoid unnecessary overhead and use sequential backend instead.
-            raise FallbackToBackend(SequentialBackend())
+            raise FallbackToBackend(
+                SequentialBackend(nesting_level=self.nesting_level))
         self.parallel = parallel
         self._n_jobs = n_jobs
         return n_jobs
@@ -421,7 +421,8 @@ class MultiprocessingBackend(PoolManagerMixin, AutoBatchingMixin,
         """Build a process or thread pool and return the number of workers"""
         n_jobs = self.effective_n_jobs(n_jobs)
         if n_jobs == 1:
-            raise FallbackToBackend(SequentialBackend())
+            raise FallbackToBackend(
+                SequentialBackend(nesting_level=self.nesting_level))
 
         already_forked = int(os.environ.get(self.JOBLIB_SPAWNED_PROCESS, 0))
         if already_forked:
@@ -462,7 +463,8 @@ class LokyBackend(AutoBatchingMixin, ParallelBackendBase):
         """Build a process executor and return the number of workers"""
         n_jobs = self.effective_n_jobs(n_jobs)
         if n_jobs == 1:
-            raise FallbackToBackend(SequentialBackend())
+            raise FallbackToBackend(
+                SequentialBackend(nesting_level=self.nesting_level))
 
         self._workers = get_memmapping_executor(
             n_jobs, timeout=idle_worker_timeout,

--- a/joblib/test/test_parallel.py
+++ b/joblib/test/test_parallel.py
@@ -1373,10 +1373,15 @@ def test_nested_parallelism_limit(backend):
     with parallel_backend(backend, n_jobs=2):
         backend_types_and_levels = _recursive_backend_info()
 
+    if cpu_count() == 1:
+        second_level_backend_type = 'SequentialBackend'
+    else:
+        second_level_backend_type = 'ThreadingBackend'
+
     top_level_backend_type = backend.title() + 'Backend'
     expected_types_and_levels = [
         (top_level_backend_type, 0),
-        ('ThreadingBackend', 1),
+        (second_level_backend_type, 1),
         ('SequentialBackend', 2),
         ('SequentialBackend', 3)
     ]

